xrayvision/visibility.py

```diff
@@ -336,11 +338,49 @@ def phase_uncertainty(self):
                     * apu.rad
                 ).to(apu.deg)
 
+    def index_by_label(self, labels: Iterable[str]):
+        """
+        Extract visibilities based on their labels.
+
+        Parameters
+        ----------
+        labels :
+            The labels of the desired visibilities.
+
+        Returns
+        -------
+        new_vis : Same as self type
+        """
+        self_labels = self.meta.vis_labels
+        if self_labels is None:
+            raise ValueError("self.meta.vis_labels must be set to index by label.")
+        idx = [np.where(self_labels == label)[0][0] for label in labels]
+        new_data = self._data.isel({self._uv_key: idx})
+        new_data.attrs[self._meta_key][_VIS_LABELS_KEY] = labels
+        new_vis = copy.deepcopy(self)
+        new_vis._data = new_data
+        return new_vis
+
     def _build_quantity(self, label, unit_label=None):
         if unit_label is None:
             unit_label = label
         return apu.Quantity(self._data[label], unit=self._data.attrs[self._units_key][unit_label])
 
+    def __getitem__(self, item):
+        if isinstance(item, numbers.Integral):
+            item = (slice(item, item + 1),)
```

Comment:
So this would return one a new vis object with length 1 in vis axis `vis[0:0]`? I think normal slicing would return an empty object `a = [0,1,2]`, `a[0:0]` gives `[]`

Reply:
An integer item is equivalent to `[0:1][0]`, not `[0:0]`.  So the right values should be given, even if the length of the data is 1 rather than 0.

I put this line in because I've used and `xarray.Dataset` as the underlying implementation which doesn't support scalar data.  If we don't like this implementation, we could either:
* Not support integer slicing and raise an error if the user provides an integer;
* Create a new `Visibility` class that gives a scalar value for all attributes.  It would have to use different underlying data storage implementation.  Potentially, it could be based on `SimpleNamespace`.  This feels like it would be most intuitive, but I question whether this would ever be useful in a scientific workflow.  Still, it shouldn't be hard to implement.
* Return a tuple of scalar values of all the attributes, e.g. `(self.isibilities[0], self.u[0], self.v[0],...)`.  This would be simple but unintuitive.
* Have this return a scalar quantity of the visibility value, i.e. equivalent to `my_vis.visibilities[0]`.

Reply:
I decided to raise an error if the slicing input resulted in a scalar visibility.  This behaviour can be loosened in the future if we want without any breaking consequences.